### PR TITLE
Add script for preparing translations

### PR DESCRIPTION
**Pull request by dacook**
#### What? Why?
I couldn't help it

#### What should a releaser test?
Run the prepare script. It will download translations, commit and push them. Then it will create a release tag and give you a shortcut for drafting the release and creating the release issue!!!

⚠️  This commits to master. The easiest way to test is to wait until next release cycle :)

#### Documentation updates
We can update the wiki

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -9,23 +9,12 @@ assignees: ''
...
-  </details>
-- [ ] Create a tag:
-    - `script/tag_release` will auto increment patch version, otherwise
+- [x] Merge pull requests in the [Ready To Go] column
+- [x] Include translations: `script/release/udpate_locales`
+- [x] Create a tag:
+    - `script/release/tag` will auto increment patch version, otherwise
     - `git push upstream HEAD:refs/tags/vX.Y.Z`
-- [ ] [Draft new release]. Look at previous [releases] for inspiration.
+- [x] [Draft new release]. Look at previous [releases] for inspiration.
```
Comment (rioug) [inline .github/ISSUE_TEMPLATE/release.md:17]: I am a bit wary of pre checking these items, technically they aren't done till you run the script. And also I believe the script doesn't draft the release, it just gives you the link to start drafting.

Comment (dacook) [reply .github/ISSUE_TEMPLATE/release.md:17]: Fair point, I was just looking to remove as many clicks as possible from the process.

Although, it's worth noting this template doesn't _create_ the issue, it only opens a draft. So technically it's not marked 'done' until you click 'create' to confirm 😉 .

Comment (mkllnk) [inline .github/ISSUE_TEMPLATE/release.md:17]: I wouldn't pre-tick any of these items either. I would combine all items covered by the script into one task though. So then it's only one tick.

There are two flows now. You can either create the issue through Github or run the script first and then follow the link to create the issue. The latter has the gotcha that you need to remember to merge Ready To Go PRs first.

The other question is: what do we do with minor and major releases? You often only know that ones you draft the release notes which happens after the tagging. Do we then remove the tag again and create a new one?

**Review by mkllnk — COMMENTED** [review]
This is really cool. So great to see that our processes are improving all the time.

File: .github/ISSUE_TEMPLATE/release.md (hunk 1 above)
Comment (dacook) [reply .github/ISSUE_TEMPLATE/release.md:17]: > The latter has the gotcha that you need to remember to merge Ready To Go PRs first.

I think we could add a reminder at the start of the the script for that. 
But..

> The other question is: what do we do with minor and major releases? You often only know that ones you draft the release notes which happens after the tagging. Do we then remove the tag again and create a new one?

That's a good question. Perhaps I'm automating too much. I think for now, I'll remove the shortcut link for creating the issue, until we sort that out.

Comment (dacook) [reply .github/ISSUE_TEMPLATE/release.md:17]: I have to admit, I'm not always thinking about what version level it should be. Maybe we could introduce more explicit rules. Eg if a PR with migration deleting data is tagged as `data destructive`, we could systematically say that means it's a minor increment. 🤔 

But I think we don't want to spend time thinking more about it..

**Comment by dacook** [comment]
I've updated the PR to not pre-tick the checkboxes, as we still want to manually process these.

File: .github/ISSUE_TEMPLATE/release.md (hunk 1 above)
Comment (mkllnk) [reply .github/ISSUE_TEMPLATE/release.md]: >  I'm not always thinking about what version level it should be.

Me neither. I looked at past releases before but forgot the result again. So this time I documented it:
https://community.openfoodnetwork.org/t/release-numbering/105/6

**Review by mkllnk — APPROVED** [review]

**Comment by dacook** [comment]
> Me neither. I looked at past releases before but forgot the result again. So this time I documented it:
https://community.openfoodnetwork.org/t/release-numbering/105/6

Awesome, thanks! What do you think about this description?
Screen Shot 2023-11-10 at 12 14 12 pm

**Comment by rioug** [comment]
When I did my first release I asked Filipe about the release number, and he said unless agreed before hand (during delivery circle for instance) we just update the patch number. This is better, I like the guidelines.

**Review by rioug — APPROVED** [review]

**Comment by rioug** [comment]
@filipefurtad0 FYI you'll be the one testing this

File: .github/ISSUE_TEMPLATE/release.md [hunk 2]
```diff
@@ -10,21 +10,11 @@ assignees: ''
...
-    </pre>
-  </details>
-- [ ] Create a tag:
-    - `script/tag_release` will auto increment patch version, otherwise
-    - `git push upstream HEAD:refs/tags/vX.Y.Z`
+- [ ] Include translations: `script/release/udpate_locales`
+- [ ] Increment version number: `git push upstream HEAD:refs/tags/vX.Y.Z`
+    - Major: if server changes are required (eg. provision with ofn-install)
+    - Minor: larger change that is irreversible (eg. migration deleting data)
+    - Patch: all others. Shortcut: `script/release/tag`
```
Comment (mkllnk) [inline .github/ISSUE_TEMPLATE/release.md:17]: Awesome!

File: script/release/update_locales [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+echo "\n*** Checking out latest master... ***\n"
+git checkout master
+git pull upstream master
+
+echo "\n*** Downloading latest Transifex translations... ***\n"
+tx pull --force
+git commit -a -m "Update all locales with the latest Transifex translations"
+
+echo "\n*** Pushing to master... ***\n"
+git push upstream master
```
Comment (filipefurtad0) [inline script/release/update_locales:23]: Maybe this is just me (as a newb) being too cautious - but I have to say I'm a bit weary of having a script push to master :see_no_evil: 

I usually run git status just to make sure I push only transifex files...

Well, let's see what happens! :smile:
(these really do sound like famous last words)

**Comment by filipefurtad0** [comment]
This message was left unsent, I wanted to provide feedback earlier:

This worked like a charm. Ran:

`./script/release/prepare `

After being prompted for my GH credentials, all went on as expected:
pushed the translations to master and opened created the draft release:

image

Thanks so much @dacook :pray:

File: script/release/prepare [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+# Execute preparation tasks for a regular patch release. Requires admin permission on the repo.
+#
+set -e
+
+# todo: ask to confirm, and remind to check the ready to go column.
```
Comment (filipefurtad0) [inline script/release/prepare:7]: Agree this is a nice to have ;-)

File: script/release/update_locales (hunk 3 above)
Comment (dacook) [reply script/release/update_locales:23]: Yes it is a little scary to automate it, but after having done it manually so many times it seemed ok.
I did actually include a `git status` check at the start of this script, which aborts if the working directory is not clean, so hopefully that's all covered.

The script still makes a few assumptions (that you have a remote called `upstream`) but hopefully it's safe to assume.

**Pull request merged**